packages/Microsoft.Bot.Components.Telephony/Actions/CallTransferDialog.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    public class CallTransferDialog : Dialog
+    {
+        [JsonConstructor]
+        public CallTransferDialog([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("$kind")]
+        public const string Kind = "CallTransferDialog";
+
+        [JsonProperty("targetPhoneNumber")]
+        public StringExpression TargetPhoneNumber { get; set; }
+
+        public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var targetPhoneNumber = TargetPhoneNumber.GetValue(dc.State);
+
+            await dc.Context.SendActivityAsync($"Transferring to \"{targetPhoneNumber}\"...");
+
+            // Create handoff event, passing the phone number to transfer to as context.
+            var poContext = new { TargetPhoneNumber = targetPhoneNumber };
+            var poHandoffEvent = EventFactory.CreateHandoffInitiation(dc.Context, poContext);
+
+            try
+            {
+                await dc.Context.SendActivityAsync(poHandoffEvent, cancellationToken);
+                await dc.Context.SendActivityAsync($"Call transfer initiation succeeded");
```

Comment:
This is not a good idea. Text should not be hardcoded (what about other languages? Also, would composer users want to control what activities are sent or other actions that are performed? In that case, this dialog would be drastically different. We can talk in the design call :)

Reply:
sounds good

Reply:
Check https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs#L105

However, I would like to see a scenario design for this. Do we really want to send an activity here?